### ISSUE TITLE
Signing out should reset the authentication token

### ISSUE THREAD

**Issue by gonzalo-bulnes** [gonzalo-bulnes/simple_token_authentication#34]
I've opened this issue to follow the discussion started by @dmdeller and @austinpray ([original post](https://github.com/gonzalo-bulnes/simple_token_authentication/issues/26#issuecomment-36264836)) while keeping it separated from the #26 issue main topic.

**Comment by TrevorHinesley** [comment]
@gonzalo-bulnes sorry if I've misunderstood (I've read all of the related issues)--what is the proper way to sign out? I don't see anything in the source allowing for sign out--so is that just up to the client? In that case, seems like a third party could just stay logged in forever.

I KNOW I'm missing here, so excuse my obvious ignorance.

**Comment by gonzalo-bulnes** [comment]
Hi @TrevorHinesley,

Unless you're explicitly using the **sign_in_token** option (typical use case: a sign in link sent in an email), using valid authentication credentials (by default authentication token + email) does not create a persistent session.

That means that the API clients have to provide the token in every request. Of couse, if a third party does get access to your credentials, they can impersonate you. To prevent that, HTTPS should **always** be used when authenticating using tokens.

That being said, nothing prevents you from [re-generating the authentication tokens regularly](https://github.com/gonzalo-bulnes/simple_token_authentication/tree/v1.15.0#tokens-generation). The good part here is that they are much less painful to renew than the user's passwords (the users would need to sign in again using their client to get the new tokens, but that's all).

Does this make sense?

**Comment by TrevorHinesley** [comment]
Ah yes that makes sense. So "sign out" on the client side would be just that--don't send the email and auth token with further requests. Correct?

**Comment by gonzalo-bulnes** [comment]
Correct. A well-behaved client would also forget the authentication token, since it will get it back (the same or a new one) when the user signs in again.

**Comment by TrevorHinesley** [comment]
Awesome, thanks so much @gonzalo-bulnes. Very helpful.
